### PR TITLE
[DOCS] Augments cat transforms API

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/53737 and https://github.com/elastic/elasticsearch/pull/53643

This PR adds missing details for the cat transforms API.

Preview: http://elasticsearch_53776.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cat-transforms.html

### CODE REVIEW

File: docs/reference/cat/transforms.asciidoc [hunk 1]
```diff
@@ -6,23 +6,167 @@
...
+
+`pipeline`, `p`:::
+(Default)
+The {transform} pipeline.
+
+`processed_documents_exp_avg`, `pdea`:::
+include::{docdir}/rest-api/common-parms.asciidoc[tag=exponential-avg-documents-processed]
+
+`reason`, `r`:::
+The reason.
```
Comment (benwtrent) [inline docs/reference/cat/transforms.asciidoc]:
```suggestion
The reason for the current state.
```

File: docs/reference/cat/transforms.asciidoc [hunk 2]
```diff
@@ -6,23 +6,167 @@
...
+`description`, `d`:::
+(Default)
+The description of the {transform}.
+
+`dest_index`, `di`, `destIndex`:::
+(Default)
+include::{docdir}/rest-api/common-parms.asciidoc[tag=dest-index]
+
+`document_total`, `dt`:::
+The total number of documents.
```
Comment (lcawl) [inline docs/reference/cat/transforms.asciidoc]: > document_total

I don't see an equivalent stat in the "get transform stats" API.  Is it based on documents_processed or documents_indexed?

File: docs/reference/cat/transforms.asciidoc [hunk 3]
```diff
@@ -6,23 +6,167 @@
...
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index-time-ms]
+
+`index_total `, `it`:::
+include::{docdir}/rest-api/common-parms.asciidoc[tag=index-total]
+
+`indexed_documents_exp_avg`, `idea`:::
+include::{docdir}/rest-api/common-parms.asciidoc[tag=exponential-avg-documents-indexed]
+
+`invocation_total`, `itotal`:::
+The total number of invocations.
```
Comment (lcawl) [inline docs/reference/cat/transforms.asciidoc]: > invocation_total

Is this related to continuous transforms? I couldn't see a matching item in the "get transform stats" API.

File: docs/reference/cat/transforms.asciidoc [hunk 4]
```diff
@@ -6,23 +6,167 @@
...
+
+`pipeline`, `p`:::
+(Default)
+The {transform} pipeline.
+
+`processed_documents_exp_avg`, `pdea`:::
+include::{docdir}/rest-api/common-parms.asciidoc[tag=exponential-avg-documents-processed]
+
+`reason`, `r`:::
+The reason for the current state.
```
Comment (lcawl) [inline docs/reference/cat/transforms.asciidoc]: > reason

Does this information appear only when the state="failed"?

File: docs/reference/cat/transforms.asciidoc (hunk 2 above)
Comment (hendrikmuhs) [reply docs/reference/cat/transforms.asciidoc]: according to the PR it is `documents_processed`

File: docs/reference/cat/transforms.asciidoc (hunk 3 above)
Comment (hendrikmuhs) [reply docs/reference/cat/transforms.asciidoc]: this is called `trigger_count` in stats

File: docs/reference/cat/transforms.asciidoc (hunk 4 above)
Comment (hendrikmuhs) [reply docs/reference/cat/transforms.asciidoc]: in the json output of get transform stats its only returned if set, which is only the case of a failure.

On the tabular output of `_cat` it can't be omitted, so it's in the table but usually empty

**Comment by hendrikmuhs** [comment]
I think you made some fair points that the naming is inconsistent for the REST layer, the HLRC layer and the `_cat` layer. I think we should fix this.
